### PR TITLE
 test: add coverage for bitcoin-cli -rpcwait

### PR DESCRIPTION
**Pull request by jonatack**
and other coverage improvements in `interface-bitcoin_cli.py`

### CODE REVIEW

File: test/functional/interface_bitcoin_cli.py [hunk 1]
```diff
@@ -55,31 +46,50 @@ def run_test(self):
...
+
+        self.log.info("Test -version with node stopped")
+        cli_response = self.nodes[0].cli("-version").send_cli()
+        assert "{} RPC client version".format(self.config['environment']['PACKAGE_NAME']) in cli_response
+
+        self.log.info("Test -rpcwait option waits for RPC connection instead of failing")
+        # Start node without RPC connection.
+        self.nodes[0].start()
+        # Verify failure without -rpcwait.
+        assert_raises_process_error(1, "Could not connect to the server", self.nodes[0].cli('getblockcount').echo)
```
Comment (MarcoFalke) [inline test/functional/interface_bitcoin_cli.py:89]: Isn't this racy?

Comment (jonatack) [reply test/functional/interface_bitcoin_cli.py:89]: Will run it in a loop and see if it's flakey.

Comment (jonatack) [reply test/functional/interface_bitcoin_cli.py:89]: Running `(for i in {1..1000}; do test/functional/interface_bitcoin_cli.py ; done)` to see.

Comment (MarcoFalke) [reply test/functional/interface_bitcoin_cli.py:89]: We generally only see races in travis or valgrind

Logically, I don't see how the rpc connection is guaranteed to be down when the cli is called here.

File: test/functional/interface_bitcoin_cli.py [hunk 2]
```diff
@@ -25,15 +25,6 @@ def run_test(self):
...
 
-        cli_response = self.nodes[0].cli("-version").send_cli()
-        assert "{} RPC client version".format(self.config['environment']['PACKAGE_NAME']) in cli_response
-
-        self.log.info("Compare responses from getwalletinfo RPC and `bitcoin-cli getwalletinfo`")
-        if self.is_wallet_compiled():
-            cli_response = self.nodes[0].cli.getwalletinfo()
-            rpc_response = self.nodes[0].getwalletinfo()
-            assert_equal(cli_response, rpc_response)
-
```
Comment (kristapsk) [inline test/functional/interface_bitcoin_cli.py:36]: Why this test removed?

Comment (jonatack) [reply test/functional/interface_bitcoin_cli.py:36]: Not removed; moved to line 75 `assert_equal(self.nodes[0].cli.getwalletinfo(), wallet_info)`

Comment (jonatack) [reply test/functional/interface_bitcoin_cli.py:36]: which avoids calling rpc getwalletinfo twice, a conditional, and a few lines of code.

File: test/functional/interface_bitcoin_cli.py (hunk 1 above)
Comment (jonatack) [reply test/functional/interface_bitcoin_cli.py:89]: Makes sense: it ran 1000 times without fail for me in macOS and Debian, but yeah, travis and valgrind. I'll wait to see the travis result for the fun of it and then probably remove that line; we'll still have the assertion with -rpcwait

File: test/functional/interface_bitcoin_cli.py (hunk 2 above)
Comment (kristapsk) [reply test/functional/interface_bitcoin_cli.py:36]: Ok, yes, I didn't read the whole block correctly, assumed it checks `-version`, which is also checked below, just that it outputs, not what it outputs. And it's `bitcoin-cli` version anyways, can differ from version returned by `-getinfo`, nothing to compare against there.

File: test/functional/interface_bitcoin_cli.py (hunk 1 above)
Comment (jonatack) [reply test/functional/interface_bitcoin_cli.py:89]: Travis all green. Maybe leave it in and remove if we see issues later?

File: test/functional/interface_bitcoin_cli.py (hunk 2 above)
Comment (jonatack) [reply test/functional/interface_bitcoin_cli.py:36]: @kristapsk thanks for reviewing! Can you ACK?

**Review by kristapsk — APPROVED** [review]
ACK c28c7b882b40cc75989dd102414de3b3dad69706

**Pull request merged**